File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int cash = -1;
    int counter = 0;

    // Loop for input
    while (cash <= 0)
    {
        cash = get_int("Change owed: ");
    }
    while (cash > 0)
    {
        if (cash - 25 >= 0)
        {
            cash = cash - 25;
            counter++;
        }
        else if (cash - 10 >= 0)
        {
            cash = cash - 10;
            counter++;
        }
        else if (cash - 5 >= 0)
        {
            cash = cash - 5;
            counter++;
        }
        else if (cash - 1 >= 0)
        {
            cash = cash - 1;
            counter++;
        }
    }
    printf("%i\n", counter);
}

File: sentimental-hello/cash.py
from cs50 import get_float


def main():
    try:
        amountGiven = get_float("Change: ")
        if (amountGiven > 0):
            # multiply to get away from float
            count(amountGiven * 100)
        else:
            main()
    # catch non-float input
    except ValueError:
        print(main())


def count(amount):

    # initialize coin-amount and values per coin
    a = 0
    coins = [25, 10, 5, 1]

    # run this loop until running out of change
    while amount > 0:
        for coin in coins:
            if amount - coin >= 0:
                amount -= coin
                a += 1
                break

    print(int(a))


main()
